### PR TITLE
fix: all hydration errors

### LINKED ISSUE #6294: Hydration errors on every page
### Summary

image

The errors are:

> Hydration failed because the initial UI does not match what was rendered on the server.
>
> There was an error while hydrating. Because the error happened outside of a Suspense boundary, the entire root will switch to client rendering.

Perhaps something is using the naive `typeof window !== "undefined"` type of check, instead of `useEffect`?

### URL

https://developer.mozilla.org/en-US/docs/Web/API/Navigator

And every other page that I visited

### Reproduction steps

Go to that URL; open browser console.

### Expected behavior

No errors

### Actual behavior

Hydration errors

### Device

Desktop

### Browser

Chrome

### Browser version

Stable

### Operating system

Mac OS

### Screenshot

_No response_

### Anything else?

_No response_

### Validations

- [X] I have read the [Community Participation Guidelines](https://www.mozilla.org/en-US/about/governance/policies/participation/).
- [X] I have verified that there isn't [already an issue](https://github.com/mdn/yari/issues) that reports the same bug to avoid creating a duplicate.
- [X] I have checked that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mdn/yari/discussions).

### PR DESCRIPTION
**Pull request by OnkarRuikar**
Fixes #6294.

The PR combines:
- #6380
- #6394
- #6401
- #6402

### CODE REVIEW

**Comment by caugner** [comment]
🔃 Triggered a [dev build](https://github.com/mdn/yari/actions/runs/2410060680), which will deploy to https://pr6403.content.dev.mdn.mozit.cloud/.

**Comment by caugner** [comment]
> 🔃 Triggered a [dev build](https://github.com/mdn/yari/actions/runs/2410060680), which will deploy to https://pr6403.content.dev.mdn.mozit.cloud/.

I'm not seeing any hydration errors on that deployment. 🤩

File: ssr/webpack.config.js [hunk 1]
```diff
@@ -46,8 +46,10 @@ module.exports = {
           {
             loader: "@svgr/webpack",
             options: {
-              svgo: true,
-              titleProp: true,
+              expandProps: "end",
+              // these properties cause hydration errors when enabled
+              svgo: false,
+              titleProp: false,
```
[the remark this answers is not in the document]
Comment (OnkarRuikar) [reply ssr/webpack.config.js:52]: The `titleProp` picks title from the React component definitions: `<MDNLogo title={title} />`
Our SVGs have hardcoded title tags. As the property was true it was stripping titles from the SVGs sent by SSR, so React was complaining during hydration saying there's a title mismatch.
Setting it to false keeps the hardcoded titles in the SVGs thus resolves the hydration issue.

We need to choose between hardcoded titles `<title>MDN Logo</title>` and title property `<MDNLogo title={'MDN Logo'} />`. The same rule needs to be followed for all SVGs. We can not mix and match!

***
Currently, we are optimizing SVGs on every request. Why not host the optimized SVGs in the first place and avoid `svgo` optimizations at runtime?
The better solution would be to optimize all the SVGs(using same svgo) at _build time_ for both client build and ssr build.  That way SVGs will be same and optimized on both sides. And it'll reduce some load on the server.

File: client/src/ui/organisms/footer/index.tsx [hunk 2]
```diff
@@ -19,7 +19,7 @@ export function Footer() {
       <div className="page-footer-grid">
         <div className="page-footer-logo-col">
           <a href="/" className="mdn-footer-logo" aria-label="MDN homepage">
-            <MDNLogo />
+            <MDNLogo aria-labelledby="mdn-footer-logo-svg" />
```
[the remark this answers is not in the document]
Comment (OnkarRuikar) [reply client/src/ui/organisms/footer/index.tsx:22]: Yes, this was also causing trouble during hydration.
The SVGR is stripping extra attributes on the root(\<svg\>) tag (even after we disable `svgo`). So had to re-provide it here.
If we want to have aria-labels on titles then we need to consider how we use above `titleProp` in SVGR.

File: ssr/webpack.config.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (caugner) [reply ssr/webpack.config.js:52]: Now that we have ejected react-scripts, we can find the client-side svgr config here:

https://github.com/mdn/yari/blob/0057d717482d9b1c9c0dd1c4d25c8c9e830edf58/client/config/webpack.config.js#L378-L391

Could we just copy those options over to `ssr/webpack.config.js` for now? In particular, this has `svgo: false`, but `titleProp: true`.

**Comment by OnkarRuikar** [comment]
@caugner sorry for the late response. I didn't get any notifications from GitHub regarding yari. Ping me in the chat whenever you need me.

Making svgr configs same for both SSR and client doesn't give any error. I've committed the changes. 
Also, with current client config `aria-labelledby="mdn-footer-logo-svg"` is not being rendered on both ssr and client side.

**Comment by caugner** [comment]
@OnkarRuikar Thank you! Something went wrong in the rebase here, could you please try and `git rebase -i upstream/main`, dropping all unrelated commits?

**Comment by OnkarRuikar** [comment]
> could you please try and `git rebase -i upstream/main`, dropping all unrelated commits?

Thanks! I've resolved it now. 😥

**Review by caugner — APPROVED** [review]

**Pull request merged**